basestar-database-server/src/main/java/io/basestar/database/action/CreateAction.java

```diff
@@ -88,29 +90,35 @@ public Result after(final ValueContext valueContext, final Context expressionCon
         // Id in path overrides id specified in body
         final String requestedId = options.getId() == null ? Instance.getId(initial) : options.getId();
 
-        final String actualId;
-        if (schema.getId() != null) {
-            actualId = schema.getId().evaluate(requestedId, expressionContext.with(CommonVars.VAR_THIS, initial));
+        final CompletableFuture<String> actualIdFuture;
+        if (schema.getSequence() != null) {
```

Comment:
This is where a formatted sequence generated value is set as the primary key of a new object